src/cpr_data_access/models.py

```diff
@@ -1097,3 +1105,105 @@ def to_huggingface(
         )
 
         return huggingface_dataset
+
+    def from_huggingface(
```

Comment:
This is a more generic question: is there any reason these factory methods are not classmethods? I always try to initialise the dataset by e.g. `Dataset.load_from_local()` instead of `Dataset().load_from_local()` which for me is less intuitive and pythonic. (I've just noticed that the documentation of Dataset also recommends the former)

Reply:
Whoops, the documentation needs updating. It used to be `Dataset.method` but with the introduction of multiple document models it made sense to me to initialise a dataset with only one document model – that way the idea of a dataset only containing one document model is baked in from the start.

Happy to consider alternatives but maybe not one for this PR - will update the docs though!

Reply:
docs updated